aws/resource_aws_glue_catalog_crawler_test.go

```diff
@@ -0,0 +1,276 @@
+		Providers: testAccProviders,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccGlueCrawlerConfigBasic,
+				Check: resource.ComposeTestCheckFunc(
+					checkGlueCatalogCrawlerExists(name, "test-basic"),
+					resource.TestCheckResourceAttr(name, "name", "test-basic"),
+					resource.TestCheckResourceAttr(name, "database_name", "test_db"),
+					resource.TestCheckResourceAttr(name, "role", "AWSGlueServiceRole-tf"),
+				),
+			},
+		},
+	})
+}
+
+func TestAccAWSGlueCrawler_jdbcCrawler(t *testing.T) {
+	const name = "aws_glue_catalog_crawler.test"
+	resource.Test(t, resource.TestCase{
+		PreCheck:  func() { testAccPreCheck(t) },
+		Providers: testAccProviders,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccGlueCrawlerConfigJdbc,
+				Check: resource.ComposeTestCheckFunc(
+					checkGlueCatalogCrawlerExists(name, "test-jdbc"),
+					resource.TestCheckResourceAttr(name, "name", "test-jdbc"),
+					resource.TestCheckResourceAttr(name, "database_name", "test_db"),
+					resource.TestCheckResourceAttr(name, "role", "AWSGlueServiceRoleDefault"),
+					resource.TestCheckResourceAttr(name, "jdbc_target.#", "1"),
+				),
+			},
+		},
+	})
+}
+
+func TestAccAWSGlueCrawler_customCrawlers(t *testing.T) {
+	const name = "aws_glue_catalog_crawler.test"
+	resource.Test(t, resource.TestCase{
+		PreCheck:  func() { testAccPreCheck(t) },
+		Providers: testAccProviders,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccGlueCrawlerConfigCustomClassifiers,
+				Check: resource.ComposeTestCheckFunc(
+					checkGlueCatalogCrawlerExists(name, "test_custom"),
+					resource.TestCheckResourceAttr(name, "name", "test_custom"),
+					resource.TestCheckResourceAttr(name, "database_name", "test_db"),
+					resource.TestCheckResourceAttr(name, "role", "tf-glue-service-role"),
+					resource.TestCheckResourceAttr(name, "table_prefix", "table_prefix"),
+					resource.TestCheckResourceAttr(name, "schema_change_policy.0.delete_behavior", "DELETE_FROM_DATABASE"),
+					resource.TestCheckResourceAttr(name, "schema_change_policy.0.update_behavior", "UPDATE_IN_DATABASE"),
+					resource.TestCheckResourceAttr(name, "s3_target.#", "2"),
+					resource.TestCheckResourceAttr(name, "s3_target.#", "2"),
+				),
+			},
+		},
+	})
+}
+
+func checkGlueCatalogCrawlerExists(name string, crawlerName string) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		rs, ok := s.RootModule().Resources[name]
+		if !ok {
+			return fmt.Errorf("not found: %s", name)
+		}
+
+		if rs.Primary.ID == "" {
+			return fmt.Errorf("no ID is set")
+		}
+
+		glueConn := testAccProvider.Meta().(*AWSClient).glueconn
+		out, err := glueConn.GetCrawler(&glue.GetCrawlerInput{
+			Name: aws.String(crawlerName),
+		})
+
+		if err != nil {
+			return err
+		}
+
+		if out.Crawler == nil {
+			return fmt.Errorf("no Glue Crawler found")
+		}
+
+		return nil
+	}
+}
+
+const testAccGlueCrawlerConfigBasic = `
+	resource "aws_glue_catalog_database" "test_db" {
+  		name = "test_db"
```

Comment:
For all resources requiring a name, we should randomize it due to:
* Acceptance test coding errors leaving dangling resources
* API errors leaving dangling resources

Generally we achieve this by adding something like the following in the acceptance test function:

```go
rName := acctest.RandomWithPrefix("tf-acc-test")
```

Then we change these test configurations to be a function themselves that call `fmt.Sprintf()`, e.g.

```go
func testAccGlueCrawlerConfigBasic(rName string) string {
  return fmt.Sprintf(`
resource "aws_glue_catalog_database" "test_db" {
  name = %q
}
# ...
`, rName)
}
```

Then back in the acceptance test function change the `Config` line to a proper function call, e.g. `Config: testAccGlueCrawlerConfigBasic(rName),`

Reply:
done